x/marker/simulation/proposals.go

```diff
@@ -79,19 +81,20 @@ func SimulateCreateSupplyIncreaseProposalContent(k keeper.Keeper) simtypes.Conte
 			return nil
 		}
 
-		newSupply := randomUint64(r, k.GetMaxTotalSupply(ctx)-k.CurrentCirculation(ctx, m).Uint64())
+		maxSupply := k.GetMaxSupply(ctx).Sub(k.CurrentCirculation(ctx, m))
+		newSupply := math.NewIntFromBigInt(math.ZeroInt().BigInt().Rand(r, maxSupply.BigInt()))
 
 		// TODO: When the simulation tests are fixed to stop breaking supply invariants through incorrect minting, the following check should be removed.
-		if newSupply > k.GetMaxTotalSupply(ctx) {
+		if newSupply.GT(k.GetMaxSupply(ctx)) {
 			println("!!!! WARNING, TOKEN SUPPLY IS INVALID, ABORTING NEW PROPOSAL !!!!")
 			return nil
 		}
 
 		return types.NewSupplyIncreaseProposal(
 			simtypes.RandStringOfLength(r, 10),
 			simtypes.RandStringOfLength(r, 100),
-			// sdk.NewCoin(m.GetDenom(), sdk.NewIntFromUint64(randomUint64(r, k.GetMaxTotalSupply(ctx)-k.CurrentCirculation(ctx, m).Uint64()))),
-			sdk.NewCoin(m.GetDenom(), sdk.NewIntFromUint64(newSupply)),
+			// sdk.NewCoin(m.GetDenom(), sdk.NewIntFromUint64(randomUint64(r, k.GetMaxSupply(ctx).Uint64()-k.CurrentCirculation(ctx, m).Uint64()))),
```

Comment:
Is there a reason to keep this around?  Not sure how long it has been commented out for...

Reply:
Nope, I can remove it.